shortest_path.cc

```diff
@@ -63,7 +65,7 @@ namespace dijkstra {
             if (e.dest == w) break;
         }
 
-        if (distances[w] >= max_dist) {
+        if (distances[w] >= max) {
```

Comment:
if you are checking for infinity here you might actually check for it explicitly by ==